crates/air_r_formatter/src/r/auxiliary/root.rs

```diff
@@ -23,4 +40,58 @@ impl FormatNodeRule<RRoot> for FormatRRoot {
             ]
         )
     }
+
+    fn is_suppressed(&self, node: &RRoot, f: &RFormatter) -> bool {
+        let Some(child) = node.expressions().first() else {
+            return false;
+        };
+
+        let comments = f.context().comments();
```

Comment:
```suggestion
        let comments = f.context().comments();

        comments.mark_suppression_checked(node.syntax());

        let Some(child) = node.expressions().first() else {
            return false;
        };
```

Even though the immediate call to `fmt_suppressed()` will recursively mark everything as suppressed, the correct thing to do here in the overrride of `is_suppressed()` is to also call `mark_suppression_checked()` on this node, since you are declaring that you are checking its suppression manually.

For an example, see TsUnionType in Biome

```rs
    fn is_suppressed(&self, node: &TsUnionType, f: &JsFormatter) -> bool {
        f.comments().mark_suppression_checked(node.syntax());

        if node.types().is_empty() {
            f.comments().is_suppressed(node.syntax())
        } else {
            // Suppression applies to first variant
            false
        }
    }
```